### PR TITLE
HPCC 22705 Enforce max value size configuration

### PR DESCRIPTION
**Pull request by rpastrana**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by rpastrana** [comment]
@wangkx please review
@dcamper  FYI

File: esp/services/ws_store/espstorelib/daliKVStore.cpp [hunk 1]
```diff
@@ -90,6 +93,10 @@ bool CDALIKVStore::set(const char * storename, const char * thenamespace, const
     if (!storetree.get())
         throw MakeStringException(-1, "DALI KV Store set(): Unable to access store '%s'", storename); //this store doesn't exist
 
+    int maxval = storetree->getPropInt("@maxValSize", 0);
```
Comment (wangkx) [inline esp/services/ws_store/espstorelib/daliKVStore.cpp]: Use DALI_KVSTORE_MAXVALSIZE_ATT?

File: esp/services/ws_store/ws_storeService.cpp [hunk 2]
```diff
@@ -79,10 +80,11 @@ void CwsstoreEx::init(IPropertyTree *_cfg, const char *_process, const char *_se
             iter->query().getProp("@name", id);
             iter->query().getProp("@type", type);
             iter->query().getProp("@description", description);
+            unsigned int maxvalsize = iter->query().getPropInt("@maxvalsize", DDFAULT_ESP_STORE_MAX_VAL_SIZE);
```
Comment (wangkx) [inline esp/services/ws_store/ws_storeService.cpp]: Use DALI_KVSTORE_MAXVALSIZE_ATT?

File: esp/services/ws_store/ws_storeService.cpp [hunk 3]
```diff
@@ -133,25 +135,34 @@ bool CwsstoreEx::onListStores(IEspContext &context, IEspListStoresRequest &req,
...
-            store->setName(iter->query().queryProp("@name"));
-            store->setCreateTime(iter->query().queryProp("@createTime"));
-            store->setType(iter->query().queryProp("@type"));
-            store->setDescription(iter->query().queryProp("@description"));
+            store->setOwner(tree->queryProp("@createUser"));
+            store->setName(tree->queryProp("@name"));
+            store->setCreateTime(tree->queryProp("@createTime"));
+            store->setType(tree->queryProp("@type"));
+            store->setDescription(tree->queryProp("@description"));
+            store->setMaxValSize(tree->queryProp("@maxValSize"));
```
Comment (wangkx) [inline esp/services/ws_store/ws_storeService.cpp]: Use DALI_KVSTORE_MAXVALSIZE_ATT?

**Review by wangkx — COMMENTED** [review]
minor

File: esp/services/ws_store/ws_storeService.cpp (hunk 3 above)
Comment (rpastrana) [reply esp/services/ws_store/ws_storeService.cpp]: Defined in another lib, not visible here.

**Comment by rpastrana** [comment]
@wangkx please review

**Review by wangkx — APPROVED** [review]
@rpastrana looks fine.

**Comment by rpastrana** [comment]
@richardkchapman please merge

**Comment by AttilaVamos** [comment]
@rpastrana These schedule related failures are not related to your PR. I will reschedule Smoketest on this PR and I am sure they will gone in next result.

**Pull request merged**